forum/qa-content/qa-page.js

```diff
@@ -870,3 +876,95 @@ function qa_ajax_error()
     }
 
 } ( document ) );
+
+
+/**
+ *	Add button, which will let users to copy link of topic/comment they are currently reading and share it
+ */
+;( function( document ) {
+
+	'use strict';
+
+	var copyToClipboard = window.initializeCopyToClipboard;
+
+    console.info( 'copyToClipboard: ', copyToClipboard, ' /boolean: ' , !!copyToClipboard );
+
+	if ( !!copyToClipboard ) {
+		var shareMethodPopup = ( function() {
+			var popup = document.createElement( 'div' );
+			popup.classList.add( 'share-popup' );
+            popup.id = 'share-popup';
+
+			var shareLinkField = document.createElement( 'input' );
+			shareLinkField.type = 'text';
+			shareLinkField.readonly = true;
+            shareLinkField.id = 'shareLinkField';
+			shareLinkField.value = '';
+
+			var copyLinkBtn = document.createElement( 'button' );
+			copyLinkBtn.type = 'button';
+            copyLinkBtn.textContent = 'Kopiuj link';
+            copyLinkBtn.classList.add( 'copy-shared-link' );
+
+			popup.appendChild( shareLinkField );
+			popup.appendChild( copyLinkBtn );
+
+			return {
+                popup: popup,
+                shareLinkField: shareLinkField
+            };
+		}() );
+
+		var addShareButton = function() {
+
+			var topicActionButtonsBar = document.querySelector( '.qa-q-view-buttons' );
+			var answersActionButtonsBar = Array.from( document.querySelectorAll( '.qa-a-item-buttons' ) );
+			var commentsActionButtonsBar = Array.from( document.querySelectorAll( '.qa-c-item-buttons' ) );
+
+			var allActionButtonsBar = answersActionButtonsBar.concat( commentsActionButtonsBar );
+			allActionButtonsBar.push( topicActionButtonsBar );
+
+			for ( var i = 0, len = allActionButtonsBar.length; i < len; i++ ) {
+				var shareButton = document.createElement( 'button' );
+                shareButton.type = 'button';
+				shareButton.textContent = 'Udostępnij';
+				shareButton.classList.add( 'share-button' );
+
+				allActionButtonsBar[ i ].appendChild( shareButton );
+			}
+
+            var popupParentRef = {};
+
+            var mainParent = document.querySelector( '.qa-main' );
+			mainParent.addEventListener( 'click', function( ev ) {
+                var eTarget = ev.target;
+                var eClassList = eTarget.classList;
+                var parentRef = eTarget.parentNode;
+
+                if ( eClassList.contains( 'share-button' ) ) {
+                    var topicContentGroup = eTarget.parentNode.classList[ 0 ].split( '-' )[ 2 ];

```

Comment:
`eTarget.parentNode.classList[ 0 ].split( '-' )[ 2 ];` - it looks like something prone to errors and tightly coupled to other components. Is there some more smart way to get that?